app/assets/stylesheets/finder_frontend.scss

```diff
@@ -268,6 +256,13 @@ mark {
   margin: 0 0 govuk-spacing(4) 0;
   border-bottom: solid 1px govuk-colour("black");
 
+  .gem-c-heading {
```

Comment:
I share the general feeling, I toyed with this one.  I didn't like directly using a gem class (in case it changed in future) but I did wonder if adding extra elements were more desirable compared to targeting via a contextually nested selector...

I can add a wrapper for this one.

Reply:
Both using a wrapper or the use of the `gem-c-heading` class will lead to the same problem - if the gem is updated, then this could potentially clash with this change in the future.

I'd suggest either not extending the component, or seeing whether this really needs to be hidden on mobile.

Reply:
> Both using a wrapper or the use of the extension of the gem-c-heading class will lead to the same problem

Ah I thought the suggestion was an additional bespoke wrapper like:

```
<div class="facets__header">
  <div class="new-wrapper-element">
  ...headingcomponent
  </div>
</div>
```
Hence worry about extra code but hooking on `govuk-heading-X` solves too.

Reply:
Yeah, both wrapping an element around the heading component or using the `gem-c-heading` class are risky for the same reason - they could clash with any future changes in the gem.

To be fair, using `@extend %govuk-heading-s;` to get all the styles for the (in this case) small heading runs into the same problem with clashes. I suppose this is both a risk - any changes to the Design System headings will be automatically changed - and a benefit - as any changes to the Design System headings will automatically be changed. I think this method is slightly more honest with its intent as it doesn't pretend to be a isolated component.

Reply:
Hopping back into this thread, having slept on it a bit, if we're having to do _anything_ to mutate the heading, is this grounds to extend the heading component or should we even be using the heading component here? I feel like trying to contend with custom app setups for headings means we're just replicating the risk we would be trying to avoid by including the heading component, no?

Reply:
Since the code has changed I can't see what it was trying to do, now. Generally I'd try to use a wrapper rather than affect a component class directly these days (I didn't in the past, but now recognise it as bad practice to be modifying the component because it's no longer a complete package). So there are levels in my mind:
best: if you need it to do something new, will others maybe want that too one day? If not is it REALLY needed or could the design be more accommodating to the current gem constraints? Ideally it's something others will want and you add the option to the gem.
less good: use a wrapper and affect that if you can
bad but maybe necessary: affect the component directly

So if I were trying to make something vanish at a certain screen size I'd probably tend towards a wrapper. If I wanted the h2 to be bright red, three times the usual size and have custom padding I'd be wondering why I'm using the header component in the first place. It's about balancing those needs because inheriting the h2 bumf provided by the component is valuable so we should if we can. If your wrapper has to target not just the wrapper but also the component class there's no value in the wrapper, but a wrapper could be used for a display:none or different padding etc, just not for overrides.

Reply:
> So if I were trying to make something vanish at a certain screen size I'd probably tend towards a wrapper. 

So in this context the show / hide reimplementation of a heading been switched from using the `gem-c` class to `govuk-heading-s` _instead_ of a wrapper.  This moves it from the gem to the Design System.  I was thinking as it's nested it's quite a contextual narrow impact without an extra wrapper (as suggested)

> If we're having to do anything to mutate the heading, is this grounds to extend the heading component or should we even be using the heading component here? I feel like trying to contend with custom app setups for headings means we're just replicating the risk we would be trying to avoid by including the heading component, no?

& 

>  is this grounds to extend the heading component or should we even be using the heading component here?

I did notice a `govuk-!-display-block` & `govuk-!-display-none` on the Design System and wondered if `@include govuk-!-display-block ` existed to use but it doesn't.  This might be a better solution to explore for future? A helper of sorts.

I did think about the heading gem having a show / hide option (per screen size) however I tend to think this should a real exception for a heading.  If we are show / hiding headings for different screen sizes often, I would lean towards there being a problem with the responsive design that requires a rethink.

>  If not is it REALLY needed or could the design be more accommodating to the current gem constraints?

Yeah this is my thoughts to not go into the gem, it's somewhat unique. Potentially we can go back and re-work the Design however not in scope for this context.

> Since the code has changed I can't see what it was trying to do, now. 

This was my worry with rebasing.  Makes me think, I should do this _after_ approval but then are you really reviewing final code...

Looking through, I can't see a couple [#1](https://github.com/alphagov/content-publisher/blob/c8538a6d60ac35cbb2c01a275b81afd3287ec060/app%2Fassets%2Fstylesheets%2Futilities%2F_overwrites.scss#L97) [#2](https://github.com/alphagov/content-publisher/blob/0dfcc98f8bd1886d7f9f6ae1c67c766030039a47/app%2Fassets%2Fstylesheets%2Fobjects%2F_history.scss#L6) of similar uses like this. 

Wrap or not to wrap. 🤔

Reply:
Added a wrapper feels in favour of this.